aeron-cluster/src/main/java/io/aeron/cluster/SequentialTimerService.java

```diff
@@ -0,0 +1,215 @@
+            final TimerEntry entry = new TimerEntry(correlationId, deadline, index);
+            timers[index] = entry;
+            timerByCorrelationId.put(correlationId, entry);
+
+            shiftUp(timers, index);
+        }
+    }
+
+    public boolean cancelTimerByCorrelationId(final long correlationId)
+    {
+        final TimerEntry removed = timerByCorrelationId.remove(correlationId);
+        if (null == removed)
+        {
+            return false;
+        }
+
+        for (int i = removed.index + 1; i < size; i++)
+        {
+            final TimerEntry entry = timers[i];
+            timers[i - 1] = entry;
+            entry.index--;
+        }
+
+        size--;
+        timers[size] = null;
+
+        return true;
+    }
+
+    public void snapshot(final TimerSnapshotTaker snapshotTaker)
+    {
+        final TimerEntry[] timers = this.timers;
+        for (int i = 0, numEntries = size; i < numEntries; i++)
+        {
+            final TimerEntry entry = timers[i];
+            snapshotTaker.snapshotTimer(entry.correlationId, entry.deadline);
+        }
+    }
+
+    public void currentTime(final long now)
+    {
+    }
+
+    private static void shiftAllUp(final TimerEntry[] timers, final int expiredTimers, final int size)
+    {
+        if (expiredTimers < size)
+        {
+            final int remainingTimers = size - expiredTimers;
+            System.arraycopy(timers, expiredTimers, timers, 0, remainingTimers);
+            Arrays.fill(timers, remainingTimers, size, null);
+            for (int i = 0; i < remainingTimers; i++)
+            {
+                timers[i].index -= expiredTimers;
+            }
+        }
+        else
+        {
+            Arrays.fill(timers, null);
+        }
+    }
+
+    private static void shiftUp(final TimerEntry[] timers, final int index)
+    {
+        for (int i = index; i > 0; i--)
+        {
+            final TimerEntry entry = timers[i];
+            final TimerEntry prevEntry = timers[i - 1];
+            if (entry.deadline < prevEntry.deadline)
+            {
+                timers[i - 1] = entry;
+                entry.index--;
+                timers[i] = prevEntry;
+                prevEntry.index++;
+            }
+            else
+            {
+                break;
+            }
+        }
+    }
+
+    private static void shiftDown(final TimerEntry[] timers, final int index, final int size)
+    {
+        for (int i = index, last = size - 1; i < last; i++)
+        {
+            final TimerEntry entry = timers[i];
+            final TimerEntry nextEntry = timers[i + 1];
+            if (entry.deadline >= nextEntry.deadline)
+            {
+                timers[i + 1] = entry;
+                entry.index++;
+                timers[i] = nextEntry;
+                nextEntry.index--;
+            }
+            else
+            {
+                break;
+            }
+        }
+    }
+
+    private static final class TimerEntry
```

Comment:
Make package private to avoid bridge classes.

Reply:
Done.